doc: tf-m: update thread with support information (#6893)
nrfconnect/sdk-nrf: samples/openthread/coap_client/README.rst

This sample uses :ref:`Zephyr CoAP API<zephyr:coap_sock_interface>` for communication, which is the preferred API to use for new CoAP applications.
For example usage of the native Thread CoAP API, see the :ref:`coap_server_sample` sample.

Device Firmware Upgrade extension
=================================

> This optional extension can be used to perform an over-the-air Device Firmware Upgrade (DFU).

Reviewer: ```suggestion
You can use this optional extension to perform an over-the-air Device Firmware Upgrade (DFU).
```

In this process, the device that hosts the new firmware image sends it to the CoAP Client device using `SMP over Bluetooth`_.
The :ref:`MCUboot <mcuboot:mcuboot_wrapper>` bootloader solution is then used to replace the old firmware image with the new one.

.. note::
   The Device Firmware Upgrade feature is currently supported only on the nRF52840 DK.

FEM support
===========

.. include:: /includes/sample_fem_support.txt

.. _coap_client_sample_multi_ext:

Multiprotocol Bluetooth LE extension
====================================

> This optional extension can demonstrate the OpenThread stack and :ref:`nrfxlib:softdevice_controller` working concurrently.

Reviewer: can demonstrate or demonstrates?
